### PR TITLE
Add publish workflow

### PR DESCRIPTION
**Pull request by larrybradley**
This version uses the TestPyPI server to test how it works.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  push:
+  pull_request:
+
+jobs:
+
+  test_wheel_building:
+    # This ensures that a couple of target wheels build fine in pull
+    # requests and pushes
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish.yml@v1
+    if: (github.event_name == 'push' || github.event_name == 'pull_request')
```
Comment (pllim) [inline .github/workflows/publish.yml:14]: You want to do this for all the PRs? No special label?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        # Linux wheels
+        - cp38-manylinux_i686
+        - cp39-manylinux_i686
+        - cp310-manylinux_i686
+        - cp38-manylinux_x86_64
+        - cp39-manylinux_x86_64
+        - cp310-manylinux_x86_64
+        - cp38-musllinux_x86_64
+        - cp39-musllinux_x86_64
+        - cp310-musllinux_x86_64
```
Comment (pllim) [inline .github/workflows/publish.yml:43]: Tom R been having issues with musllinux on astropy but hopefully you have better luck here. I just don't even bother with it for synphot...

**Review by pllim — COMMENTED** [review]
At a glance, LGTM. If you are in doubt, you can try ping Tom R, Stuart M, or Conor M to have a look too.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (larrybradley) [reply .github/workflows/publish.yml:43]: OK, I'll remove it if it has issues.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (larrybradley) [reply .github/workflows/publish.yml:14]: Yes, but it only does 2 wheels.  I copied this from astropy.  Perhaps it's unnecessary.

**Comment by pllim** [comment]
Good luck!

**Pull request merged**